From 792e55e997adfbd821f8e191a7b687c014c0c027 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 19 Oct 2020 18:13:23 +0800
Subject: [PATCH 31/41] HACK: xdg-shell-v6: Don't abort when client size
 mismatched

There's a race in qtwayland might causing size mismatch, let's
workaround it here for now.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston-desktop/xdg-shell-v6.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/libweston-desktop/xdg-shell-v6.c b/libweston-desktop/xdg-shell-v6.c
index 7dd6447..3a2f537 100644
--- a/libweston-desktop/xdg-shell-v6.c
+++ b/libweston-desktop/xdg-shell-v6.c
@@ -658,14 +658,8 @@ weston_desktop_xdg_toplevel_committed(struct weston_desktop_xdg_toplevel *toplev
 	if ((toplevel->next.state.maximized || toplevel->next.state.fullscreen) &&
 	    (toplevel->next.size.width != geometry.width ||
 	     toplevel->next.size.height != geometry.height)) {
-		struct weston_desktop_client *client =
-			weston_desktop_surface_get_client(toplevel->base.desktop_surface);
-		struct wl_resource *client_resource =
-			weston_desktop_client_get_resource(client);
-
-		wl_resource_post_error(client_resource,
-				       ZXDG_SHELL_V6_ERROR_INVALID_SURFACE_STATE,
-				       "xdg_surface buffer does not match the configured state");
+		weston_desktop_xdg_surface_schedule_configure(&toplevel->base);
+		weston_log("xdg_surface buffer does not match the configured state\n");
 		return;
 	}
 
-- 
2.20.1

